x-pack/plugins/siem/public/components/page/add_to_kql/index.tsx

```diff
@@ -0,0 +1,114 @@
+  children: JSX.Element;
+  expression: string;
+  filterQueryDraft: KueryFilterQuery;
+}
+
+class AddToKqlComponent extends React.PureComponent<Props> {
+  public render() {
+    const { children } = this.props;
+    return (
+      <WithHoverActions
+        hoverContent={
+          <HoverActionsContainer data-test-subj="hover-actions-container">
+            <EuiToolTip content={i18n.FILTER_TO_KQL}>
+              <EuiIcon type="filter" onClick={this.addToKql} />
+            </EuiToolTip>
+          </HoverActionsContainer>
+        }
+        render={() => children}
+      />
+    );
+  }
+
+  private addToKql = () => {
+    const { expression, filterQueryDraft, applyFilterQueryFromKueryExpression } = this.props;
+    applyFilterQueryFromKueryExpression(
+      filterQueryDraft && !isEmpty(filterQueryDraft.expression)
+        ? `${filterQueryDraft.expression} and ${expression}`
+        : expression
+    );
+  };
+}
+
+const HoverActionsContainer = styled(EuiPanel)`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+  height: 25px;
+  justify-content: center;
+  left: 5px;
+  position: absolute;
+  top: -10px;
+  width: 30px;
+  cursor: pointer;
+`;
+
+interface AddToKqlProps {
+  children: JSX.Element;
+  expression: string;
+  componentFilterType: 'network' | 'hosts';
+  type: networkModel.NetworkType | hostsModel.HostsType;
+}
+
+export const AddToKql = pure<AddToKqlProps>(
+  ({ children, expression, type, componentFilterType }) => (
+    <WithSource sourceId="default" indexTypes={[IndexType.FILEBEAT, IndexType.PACKETBEAT]}>
+      {({ indexPattern }) => {
+        switch (componentFilterType) {
+          case 'hosts':
+            return (
+              <HostsFilter indexPattern={indexPattern} type={type as hostsModel.HostsType}>
+                {({ applyFilterQueryFromKueryExpression, filterQueryDraft }) => (
+                  <AddToKqlComponent
+                    applyFilterQueryFromKueryExpression={applyFilterQueryFromKueryExpression}
+                    expression={expression}
+                    filterQueryDraft={filterQueryDraft}
+                  >
+                    {children}
+                  </AddToKqlComponent>
+                )}
+              </HostsFilter>
+            );
+          case 'network':
+            return (
+              <NetworkFilter indexPattern={indexPattern} type={type as networkModel.NetworkType}>
+                {({ applyFilterQueryFromKueryExpression, filterQueryDraft }) => (
+                  <AddToKqlComponent
+                    applyFilterQueryFromKueryExpression={applyFilterQueryFromKueryExpression}
+                    expression={expression}
+                    filterQueryDraft={filterQueryDraft}
+                  >
+                    {children}
+                  </AddToKqlComponent>
+                )}
+              </NetworkFilter>
+            );
+        }
+        throw new Error(`Unknown Filter Type in switch statement ${componentFilterType}`);
```

Comment:
In other parts of the code we have used a assert unreachable such as this example from

`x-pack/plugins/siem/server/lib/network/query_top_n_flow.dsl.ts`

```ts
assertUnreachable(networkTopNFlowType);
```

```ts
const assertUnreachable = (x: never): never => {
  throw new Error(`Unknown Field in switch statement ${x}`);
};
```

I try to put mine on the `default` explicitly because then with that `never` clause of TypeScript you will get a static type check error and not have to rely on runtime error. It's TypeScript's way of letting us use static types to deal with what other programming languages refer to as "exhaustive matching" where you must deal with all of the case statements and cannot add a new one without the type compiler complaining.

Ref: http://ideasintosoftware.com/exhaustive-switch-in-typescript/

Just pointing it out in case you wanted to use one here.